### PR TITLE
SWC-7081: pass the invitation token to One Sage

### PR DESCRIPTION
**Pull request by jay-hodgson**

### CODE REVIEW

File: src/main/java/org/sagebionetworks/web/client/place/users/RegisterAccount.java [hunk 1]
```diff
@@ -1,19 +1,17 @@
 package org.sagebionetworks.web.client.place.users;
 
-import com.google.gwt.place.shared.Place;
 import com.google.gwt.place.shared.PlaceTokenizer;
 import com.google.gwt.place.shared.Prefix;
+import org.sagebionetworks.web.client.place.ParameterizedPlace;
 
-public class RegisterAccount extends Place {
+public class RegisterAccount extends ParameterizedPlace {
```
Comment (jay-hodgson) [inline src/main/java/org/sagebionetworks/web/client/place/users/RegisterAccount.java:7]: Change to a parameterized place so we can pass both the email and the signed token to the register account page (which will be passed to One Sage)

File: src/main/java/org/sagebionetworks/web/client/place/users/RegisterAccount.java [hunk 2]
```diff
@@ -1,19 +1,17 @@
...
 
-  private String token;
+  public static final String EMAIL_QUERY_PARAM = "email";
+  public static final String MEMBERSHIP_INVTN_QUERY_PARAM =
+    "membershipInvtnSignedToken";
 
   public RegisterAccount(String token) {
```
Comment (jay-hodgson) [inline src/main/java/org/sagebionetworks/web/client/place/users/RegisterAccount.java:13]: Note that there are cases in the codebase that point to RegisterAccount with a default token ("0").  I tested that the presenter just redirects to the OneSage registration page without any additional query params (besides the appId used to brand the One Sage site)

File: src/main/java/org/sagebionetworks/web/client/presenter/users/RegisterAccountPresenter.java [hunk 3]
```diff
@@ -21,22 +21,21 @@ public void start(AcceptsOneWidget panel, EventBus eventBus) {}
...
-      Window.Location.replace(
-        WebConstants.ONESAGE_PRODUCTION_URL +
-        "/register1?" +
-        WebConstants.ONESAGE_SYNAPSE_APPID_QUERY_PARAM
-      );
+    String emailInvitationToken = place.getParam(
+      RegisterAccount.MEMBERSHIP_INVTN_QUERY_PARAM
+    );
+    String email = place.getParam(RegisterAccount.EMAIL_QUERY_PARAM);
+    StringBuilder targetUrl = new StringBuilder();
```
Comment (jay-hodgson) [inline src/main/java/org/sagebionetworks/web/client/presenter/users/RegisterAccountPresenter.java:28]: Build up the target URL based on what parameters we have

File: src/main/java/org/sagebionetworks/web/client/presenter/users/RegisterAccountPresenter.java [hunk 4]
```diff
@@ -21,22 +21,21 @@ public void start(AcceptsOneWidget panel, EventBus eventBus) {}
...
+    targetUrl.append(WebConstants.ONESAGE_PRODUCTION_URL);
+    targetUrl.append("/register1?");
+    targetUrl.append(WebConstants.ONESAGE_SYNAPSE_APPID_QUERY_PARAM);
+
+    if (emailInvitationToken != null) {
+      targetUrl.append("&signedToken=" + emailInvitationToken);
```
Comment (jay-hodgson) [inline src/main/java/org/sagebionetworks/web/client/presenter/users/RegisterAccountPresenter.java:34]: This is where the magic happens.   One Sage already has logic to read a signed token from the query parameters and store it in Local Storage.  Then on the Account Created page, it already has logic to read this token in, check if it's of this type, and even automatically joins the target team (as well as changing the link to say something like "Take me to the Team"!).  This workflow functions perfectly (I just tested it), once we send over this signed token!

**Review by nickgros — APPROVED** [review]

**Pull request merged**